renku/core/commands/workflow.py

```diff
@@ -118,3 +128,70 @@ def _create_workflow(output_file, revision, paths, client: LocalClient):
 def create_workflow_command():
     """Command that create a workflow description for a file."""
     return Command().command(_create_workflow)
+
+
+def unique_workflow(database: Database) -> Dict[str, Plan]:
+    """Map of unique plans in the provenance graph indexed by name."""
+    workflows = dict()
+    plans = database.get("plans")
+    for p in plans.values():
+        if p.invalidated_at is None and p.name not in workflows:
+            workflows[p.name] = p
+    return workflows
+
+
+@inject.autoparams()
+def _edit_workflow(
+    name,
+    database: Database,
+    new_name: Optional[str],
+    description: Optional[str],
+    set_params: Dict[str, str],
+    map_params: Optional[str],
+    rename_params: Dict[str, str],
+    describe_params: Dict[str, str],
+):
+    """Edits a workflow details."""
+    workflows = unique_workflow(database)
+    if name not in workflows.keys():
+        errors.ParameterError(f'The specified workflow is "{name}" is not an active workflow.')
+
+    workflow = workflows[name]
+    if new_name:
+        workflow.name = new_name
+
+    if description:
+        workflow.description = description
+
+    for name, value in set_params.items():
+        for param in workflow.parameters:
+            if param.name == name:
+                param.default_value = value
+                break
+        else:
+            # TODO: what's the position
```

Comment:
@Panaetius based on the RFC specs `--set` adds a new parameter in case if it doesn't exist. should the optional `position` argument for `CommandParameter` be None or rather set it to be the last parameter, based on the current parameters in the parameters list?

Reply:
`position` should be `None` if it's not in the parameter list for the shell command.